core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/Path.kt

```diff
@@ -39,7 +59,7 @@ fun buildPathFrom(
         if (totalLength >= endOffset)
             break
         val length = infra.getTrackChunkLength(dirChunkId.value)
-        if (totalLength + length >= beginOffset)
+        if (totalLength + length > beginOffset)
```

Comment:
I don't think this change is correct. I'd also suggest transforming the code as follows:

```kotlin
val blockEndOffset = totalLength + length
// if the block ends before the path starts, it can be safely skipped
if (beginOffset >= blockEndOffset) {
    mutBeginOffset -= length
    mutEndOffset -= length
} else {
    filteredChunks.add(dirChunkId)
}
```

If a block ends where the path starts, it can be skipped

Reply:
Code does the same thing in the end ;) However, if this formatting makes more sense, we can take this modification into account.